### DISCUSSION TITLE
Exclusive Mode Not Working

### DISCUSSION THREAD

**Discussion by chris027** [ExistentialAudio/BlackHole#685] [Q&A]
Whenever I tell my music playback applications to use exclusive mode for a Blackhole audio device, I get no audio. Disabling exclusive mode in the apps, makes everything work again. 

Is exclusive mode possible when outputting to Blackhole?

**Comment by chris027** [comment]
Just following up on this. 

Is it possible to use exclusive mode and Blackhole?

I can use exclusive mode with another virtual interface, just not blackhole. I'd love this feature because then the sample rate would automatically change based on the music being played. Without exclusive mode the sample rate remains the same.

**Comment by devinroth** [comment]
Well it's kind of a yes and no situation. If you use Exclusive Mode on BlackHole only one app will be able to use BlackHole so that creates a problem. The sample rate change is completely unrelated to exclusive mode unless the app that is playing the music will only update the sample rate when in exclusive mode. That's not a limitation of BlackHole rather the Apps using BlackHole. 

A potential solution is to compile a custom version of BlackHole with the mirrored device visible which would then allow two apps to use BlackHole in exclusive mode. `kDevice2_IsHidden = false` will do the trick. Then you have two identical BlackHole devices that you can use.

**Reply by chris027** [reply]
Ah, I never thought about that as I assumed the exclusive access was either on the output or input side, not both. 

Thanks for the info. This app is fantastic.
